core/src/main/java/com/sequenceiq/cloudbreak/converter/stack/cluster/ServiceEndpointCollector.java

```diff
@@ -0,0 +1,189 @@
+package com.sequenceiq.cloudbreak.converter.stack.cluster;
+
+import static com.sequenceiq.cloudbreak.common.type.OrchestratorConstants.YARN;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.sequenceiq.cloudbreak.api.model.ExposedService;
+import com.sequenceiq.cloudbreak.api.model.GatewayType;
+import com.sequenceiq.cloudbreak.api.model.Port;
+import com.sequenceiq.cloudbreak.blueprint.validation.BlueprintValidator;
+import com.sequenceiq.cloudbreak.blueprint.validation.StackServiceComponentDescriptor;
+import com.sequenceiq.cloudbreak.blueprint.validation.StackServiceComponentDescriptors;
+import com.sequenceiq.cloudbreak.domain.Blueprint;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.gateway.ExposedServices;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.gateway.Gateway;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.gateway.GatewayTopology;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.host.HostGroup;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.service.network.NetworkUtils;
+
+@Component
+public class ServiceEndpointCollector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceEndpointCollector.class);
+
+    private final BlueprintValidator blueprintValidator;
+
+    private final StackServiceComponentDescriptors stackServiceComponentDescs;
+
+    public ServiceEndpointCollector(BlueprintValidator blueprintValidator, StackServiceComponentDescriptors stackServiceComponentDescs) {
+        this.blueprintValidator = blueprintValidator;
+        this.stackServiceComponentDescs = stackServiceComponentDescs;
+    }
+
+    public Map<String, String> collectServiceUrlsForPorts(Cluster cluster, String ambariIp) {
```

Comment:
We have a public and a private method with the same name. i spent 3 minutes to figure out :D